daemon/daemon_test.go

```diff
@@ -28,3 +37,111 @@ type DaemonSuite struct {
 }
 
 var _ = Suite(&DaemonSuite{})
+
+func (ds *DaemonSuite) TestuseK8sNodeCIDR(c *C) {
```

Comment:
We can make the k8s client have a default instance like `pkg/kvstore` is done as well. That way, none of this has to be tied to the daemon and we can move all of this code into `pkg/k8s`